v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,306 @@
+    ~~~ shell
+    $ openssl genrsa -out my-safe-directory/ca.key 2048 
+    $ chmod 400 my-safe-directory/ca.key
+    ~~~
+
+4. Create the CA certificate using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create CA certificate.
+    $ openssl req \
+      -new \
+      -x509 \
+      -config ca.cnf \
+      -key my-safe-directory/ca.key \
+      -out certs/ca.crt \
+      -days 3660 \
+      -batch
+
+    # Reset database and index files.
+    $ rm -f index.txt serial.txt
+    $ touch index.txt
+    $ echo '01' > serial.txt
+    ~~~
+
+### Create the certificate and key pairs for nodes
+
+1. Create the node.cnf file for the first node:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # OpenSSL node configuration file
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    req_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = node
+
+    [ extensions ]
+    subjectAltName = DNS:<node-hostname>,DNS:<node-domain>,IP:<IP Address>
+    ~~~
+
+    {{site.data.alerts.callout_info}}The <code>commonName</code> and <code>subjectAltName</code> parameters are vital for CockroachDB functions. It is also important that <code>commonName</code> be set to <code>node</code>. You can modify or leave out other parameters as per your preferred OpenSSL configuration, but do not leave out the <code>commonName</code> and <code>subjectAltName</code> parameters.  {{site.data.alerts.end}}
+  
+
+2. Create the key for the first node using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ openssl genrsa -out certs/node.key 2048 
+    $ chmod 400 certs/node.key
+    ~~~
+
+3. Create the CSR for the first node using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create Node certificate signing request.
+    $ openssl req \
+      -new \
+      -config node.cnf \
+      -key certs/node.key \
+      -out node.csr \
+      -batch
+    ~~~
+
+4. Sign the node CSR to create the node certificate for the first node using the [`openssl ca`](https://wiki.openssl.org/index.php/Manual:Ca(1)) command. 
+
+    We recommend that you use the CockroachDB default value of the node certificate expiration period, which is 1830 days. You can set the CA certificate expiration period using the `days` flag.
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Sign the CSR using the CA key.
+    $ openssl ca \
+      -config ca.cnf \
+      -keyfile my-safe-directory/ca.key \
+      -cert certs/ca.crt \
+      -policy signing_policy \
+      -extensions signing_node_req \
+      -out certs/node.crt \
+      -outdir certs/ \
+      -in node.csr \
+      -days 1830 \
+      -batch
+    ~~~
+
```

Comment:
I missed a step here. Turns out openssl also writes the plaintext certificate (as opposed to the encoded certificate `node.crt`) to `certs/<certificate id>.pem`.
We can either:
1. change outdir to `/tmp/`
2. run `rm certs/*.pem` to delete that file

The file is not needed since it's the same as `node.crt`, just a different encoding.
It doesn't hurt to leave it, but that's just extra stuff lying around.

Reply:
So we either change outdir to tmp OR delete the file?

Reply:
Correct, we need one or the other. A third possibility is to just leave it and hope noone minds. Since we scp only the files we need, that might be fine.

Reply:
Are we sure the plaintext file won't be needed for something else in the user's workflow? There might be a reason OpenSSL creates it in the first place.

Reply:
I don't think we should change the outdir to tmp instead of certs - that will break the parallelism with the cockroach cert command doc. I prefer either removing the file or letting it be.

Reply:
The plaintext file is needed if we don't use `--out node.crt`. By default, openssl doesn't create that `node.crt` file, just the plaintext one. However, there's no option to have the encoded file without the plaintext file.
It's fine keeping it though.

Reply:
As discussed, we are keeping the file :)